app/src/main/java/space/taran/arknavigator/mvp/model/repo/kind/ResourceKindFactory.kt

```diff
@@ -60,4 +62,15 @@ object GeneralKindFactory {
                 ResourceExtra(id, entry.key.ordinal, entry.value!!)
             }
     }
+
+    private fun findFactory(path: Path): ResourceKindFactory<ResourceKind>? {
+        var factory = factories.find { it.isValid(path) }
+        if (factory != null) return (factory as ResourceKindFactory<ResourceKind>)
```

Comment:
Aha.. So we are calling Tika only if we haven't found any factory for the resource basing on its path. That's good, no need to worry about performance (I think reading first 100K characters is a bit demanding).